src/components/CandidateForm.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import styles from './CandidateForm.module.css'
+
+const CandidateForm = (props) => {
+    let defaultState = Object.assign({}, ...props.inputData.map(input => {
+        const {value, name} = input;
+        return {
+            [name]: value
+        }
+    }));
+    const [state, setState] = useState(defaultState);
+
+    const handleOnChange = (e) => {
+        e.preventDefault();
+        const {name, value} = e.target;
+        setState(prevState => {
```

Comment:
If you change the defaultState as suggested in the comment above,  you would have to set the new values of the object differently.